playground/azuresql/AzureSql.AppHost/Program.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddAzureProvisioning();
+
+var rg = builder.AddParameter("rg");
+
+var db = builder.AddBicepTemplate("sql1", "sql.bicep")
+    .WithParameter("p0", rg);
+
+builder.AddProject<Projects.AzureSql_ApiService>("api")
+       .WithEnvironment(ctx =>
+       {
+           var databaseName = "db";
+
+           if (ctx.PublisherName == "manifest")
+           {
+               ctx.EnvironmentVariables["ConnectionStrings__db"] = $"Server=tcp:{{{db.Resource.Name}.outputs.sqlServerName}}.database.windows.net,1433;Initial Catalog={databaseName};Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;Authentication=\"Active Directory Default\";";
+               return;
+           }
+
+           var hostName = db.Resource.Output["sqlServerName"];
+           
+           var cs = $"Server=tcp:{hostName}.database.windows.net,1433;Initial Catalog={databaseName};Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;Authentication=\"Active Directory Default\";";
```

Comment:
You really want the entire connection string to come from the resource provider. The reason is that the above code only works on Azure Public cloud. If you are using something like mooncake or one of the other non-public Azure clouds the endpoints for the resources will be different. I realize this is code that the end developer writes, but we would probably wrap AddBicepTemplate with more concrete resource types.

Reply:
You mean the host name, there’s no connection string property that comes from SQL.

This is a primitive, and an escape hatch 🤓